modules/tinymce/src/core/main/ts/newline/InsertBlock.ts

```diff
@@ -235,6 +242,15 @@ const addBrToBlockIfNeeded = (dom, block) => {
   }
 };
 
+const shouldEndContainer = (editor: Editor, container: Node) => {
+  const optionValue = Options.shouldEndContainerOnEmptyBlock(editor);
+  if (Type.isString(optionValue)) {
+    return Type.isNonNullable(container) && Arr.contains(optionValue.split(/, ?/), container.nodeName.toLowerCase());
+  } else {
+    return Type.isNonNullable(container) && optionValue;
+  }
+};
```

Comment:
```suggestion
const shouldEndContainer = (editor: Editor, container: Node | undefined) => {
  const optionValue = Options.shouldEndContainerOnEmptyBlock(editor);
  if (Type.isNullable(container)) {
    return false;
  } else if (Type.isString(optionValue)) {
    return Arr.contains(Tools.explode(optionValue), container.nodeName.toLowerCase());
  } else {
    return optionValue;
  }
};
```
I believe this is a little easier to read and avoids the duplication of the `Type.isNonNullable` checks. I also swapped to use `Tools.explode` as thats the more consistent API rather than manually splitting the string.